### PR TITLE
Implements hookspecs for anonymization pipeline

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Implements [ATO-1021](https://rasahq.atlassian.net/browse/ATO-1021)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1021]: https://rasahq.atlassian.net/browse/ATO-1021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/test_plugin.py [hunk 1]
```diff
@@ -43,3 +43,48 @@ def test_plugin_create_tracker_store(
     manager.hook.create_tracker_store.assert_called_once_with(
         endpoint_config=endpoint_config, domain=domain, event_broker=event_broker
     )
+
+
+@pytest.mark.parametrize("endpoints_file", [None, "test_endpoints.yml"])
```
Comment (Tawakalt) [inline tests/test_plugin.py:48]: Do we also need to test for when the file is not empty/None?

File: tests/test_plugin.py [hunk 2]
```diff
@@ -43,3 +43,48 @@ def test_plugin_create_tracker_store(
...
+    )
+
+    manager.hook.read_anonymization_rules(endpoints_file=endpoints_file)
+    manager.hook.read_anonymization_rules.assert_called_once_with(
+        endpoints_file=endpoints_file
+    )
+
+
+@pytest.mark.parametrize(
+    "anonymization_rules, event_broker_config",
```
[the remark this answers is not in the document]
Comment (ancalita) [reply tests/test_plugin.py:65]: This is trickier for `anonymization_rules` on OSS end, because we can't import the object type that would make up the `List` from `rasa-plus`. For `event_broker_config`, I'm testing with both `None` and `EndpointConfig` object type (we don't really need to pass args to this object).

[the remark this answers is not in the document]
Comment (Tawakalt) [reply tests/test_plugin.py:65]: Yeah, I figured this out when I reviewed the other PR

**Review by radovanZRasa — APPROVED** [review]
LGTM
